spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ChannelBindingServiceProperties.java

```diff
@@ -240,8 +240,12 @@ public String getBinder(String channelName) {
 	}
 
 	public BindingProperties getBindingProperties(String channelName) {
-		BindingProperties bindingProperties = bindings.containsKey(channelName) ?
-				bindings.get(channelName) : new BindingProperties();
+		if (this.bindings.containsKey(channelName)) {
+			return this.bindings.get(channelName);
+		}
+		// Update the bindings map with the new Binding properties.
+		BindingProperties bindingProperties = new BindingProperties();
```

Comment:
Given, we **only** create the BindingProperties when there is no explicit binding properties defined in the environment the `channelName` (since BindingProperties themselves boot ConfigurationProperties), this seems to be the only way to populate the default values for the BindingProperties (especially when the value is set to something like channelName). I made a change to update the BindingProperties with the destination name when the object is created here. This will at least ensure that all `ChannelNames` will have their `BindingProperties` in the map. 

What issue do you see if we do this way?